### PR TITLE
[BE]: Simplify some list comps to generators C419

### PR DESCRIPTION
**Pull request by Skylion007**
Simplifies some list comprehensions to generator which is more efficient. Automatically applied diffs for the most part with ruff

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: test/test_nestedtensor.py [hunk 1]
```diff
@@ -5231,16 +5227,12 @@ def test_jagged_layout_construction_as_nested_tensor(
             expected_batch_size = len(tensor_list)
             expected_contiguous = True
             expected_min_seqlen = min(
-                [
-                    (torch.tensor(t) if isinstance(t, list) else t).shape[0]
-                    for t in tensor_list
-                ]
+                (torch.tensor(t) if isinstance(t, list) else t).shape[0]
```
Comment (vadimkantorov) [inline test/test_nestedtensor.py:5230]: I think `torch.as_tensor(t)` would have this exact same semantics and simpler code at the same time

Comment (Skylion007) [reply test/test_nestedtensor.py:5230]: Fair, but this is a test case

Comment (vadimkantorov) [reply test/test_nestedtensor.py:5230]: Yeah, it should ideally be applied to all occurrences...

Comment (Skylion007) [reply test/test_nestedtensor.py:5230]: Makes sense for a follow up PR

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**